linera-storage/src/db_storage.rs

```diff
@@ -937,20 +970,22 @@ where
     fn deserialize_certificate(
         pair: &[Option<Vec<u8>>],
         hash: CryptoHash,
-    ) -> Result<ConfirmedBlockCertificate, ViewError> {
-        let cert_bytes = pair[0]
-            .as_ref()
-            .ok_or_else(|| ViewError::not_found("certificate bytes for hash", hash))?;
-        let value_bytes = pair[1]
-            .as_ref()
-            .ok_or_else(|| ViewError::not_found("value bytes for hash", hash))?;
+    ) -> Result<ResultReadCertificate, ViewError> {
+        let Some(cert_bytes) = pair[0].as_ref() else {
+            return Ok(Err(ReadCertificateError::MissingLiteCertificate(hash)));
+        };
+        let Some(value_bytes) = pair[1].as_ref() else {
+            return Ok(Err(ReadCertificateError::MissingLiteCertificate(hash)));
+        };
         let cert = bcs::from_bytes::<LiteCertificate>(cert_bytes)?;
         let value = bcs::from_bytes::<ConfirmedBlock>(value_bytes)?;
-        assert_eq!(value.hash(), hash);
-        let certificate = cert
+        let value_hash = value.hash();
+        if value_hash != hash {
```

Comment:
But this is something `linera-storage` itself guarantees, isn't it? Same for `with_value` below succeeding.

The other error can be caused by the user, when trying to read data that was never written.

So I'm not sure they belong in the same error type.